### PR TITLE
Issue 35

### PR DESCRIPTION
**Pull request by fuzzythecat**
의역이 넘쳐나는 글입니다. 
리뷰 부탁드립니다!

### CODE REVIEW

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 1]
```diff
@@ -0,0 +1,835 @@
...
+		if K.image_data_format() == "channels_first":
+			inputShape = (depth, height, width)
+			chanDim = 1
+```
+
+모델의 정의부입니다. `build` 함수를 통해 CNN모델을 만들 수 있습니다. 
+
+`build` 함수는 `width`, `height`, `depth`, `classes`의 총 네 가지 입력변수를 필요로 합니다.
+`depth`는 입력 이미지의 채널 수를 지정하며 `classes`는 범주/클래스의 개수(정수)입니다(클래스 라벨 자체는 아님). 
+`train.py` 스크립트에서 이러한 매개변수를 사용하여 96 x 96 x 3 입력 볼륨을 갖는 모델을 만들 것입니다.
```
Comment (mike2ox) [inline 35_케라스를 이용한 다중 라벨 분류.md]: 96 x 96 x 3 -> `96 x 96 x 3`

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+			input_shape=inputShape))
+		model.add(Activation("relu"))
+		model.add(BatchNormalization(axis=chanDim))
+		model.add(MaxPooling2D(pool_size=(3, 3)))
+		model.add(Dropout(0.25))
+```
+
+위의 합성곱 계층은 32개의 필터와 3 x 3 크기의 커널을 가지며, 최종 값은 ReLU 활성화 함수를 거치게 됩니다. 
+
+드롭아웃은 현재 계층과 다음 계층을 연결하는 노드들의 값을 무작위로 0으로 바꾸어주는 과정입니다(연결을 끊어주는 효과).
```
Comment (mike2ox) [inline 35_케라스를 이용한 다중 라벨 분류.md]: 드롭아웃 -> 드롭아웃(dropout) 표기 해놓으면 어떨까요.

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 3]
```diff
@@ -0,0 +1,835 @@
...
+		# use a *softmax* activation for single-label classification
+		# and *sigmoid* activation for multi-label classification
+		model.add(Dense(classes))
+		model.add(Activation(finalAct))
+ 
+		# return the constructed network architecture
+		return model
+```
+
+완전하게 연결된 계층인 `Dense`는 모델의 마지막에 배치됩니다.
```
Comment (mike2ox) [inline 35_케라스를 이용한 다중 라벨 분류.md]: 완전하게 연결된 계층(Fully Connected layer) 로 표기하면 어떨까요. 처음읽을땐 FC layer라고 인식하지 못했습니다.

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 4]
```diff
@@ -0,0 +1,835 @@
...
+from imutils import paths
+import numpy as np
+import argparse
+import random
+import pickle
+import cv2
+import os
+```
+
+위 코드에서는 해당 스크립트에 필요한 모듈들을 가져옵니다. `matplotlib.use("Agg")`로 렌더링을 위한 `matplotlib` 백엔드를 설정합니다. (본 튜토리얼은 Keras, scikit-learn, matplotlib, imutils 그리고 OpenCV가 모두 설치되어있다는 가정 하에 진행됩니다.)
```
Comment (mike2ox) [inline 35_케라스를 이용한 다중 라벨 분류.md]: 가정 하에 -> 가정하에

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 5]
```diff
@@ -0,0 +1,835 @@
...
+```
+
+스크립트에 대한 명령줄 인수들은 함수에 대한 매개 변수들과 같습니다. 만약 비유가 이해가 가지 않으신다면 명령줄 인수에 대한 [블로그 포스팅](https://www.pyimagesearch.com/2018/03/12/python-argparse-command-line-arguments/)을 읽어보시는 것을 추천합니다. 
+
+위 스크립트에는 총 네 가지의 인수들이 있습니다.
+
+1. `--dataset` : 데이터 셋의 경로입니다.
+2. `--model` : 디스크에 저장된 케라스 모델의 경로입니다.
+3. `--labelbin` : 학습의 결과인 `MultiLabelBinarizer`가 저장될 경로입니다.
+4. `--plot` : 학습 손실 값과 정확도 값의 그래프가 저장될 경로입니다.
```
Comment (mike2ox) [inline 35_케라스를 이용한 다중 라벨 분류.md]: 손실 값 -> 손실값

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 6]
```diff
@@ -0,0 +1,835 @@
...
+EPOCHS = 75
+INIT_LR = 1e-3
+BS = 32
+IMAGE_DIMS = (96, 96, 3)
+```
+
+위의 변수들은 다음과 같은 의미를 갖습니다.
+
+* 네트워크는 75 `EPOCHS`동안 역전파를 통해 점진적으로 패턴을 학습합니다. 
+* 초기 러닝 레이트는 `1e-3`(Adam 옵티마이저의 기본값)입니다.
```
Comment (mike2ox) [inline 35_케라스를 이용한 다중 라벨 분류.md]: 러닝 레이트 -> 학습률(learning rate)

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 7]
```diff
@@ -0,0 +1,835 @@
...
+model = SmallerVGGNet.build(
+	width=IMAGE_DIMS[1], height=IMAGE_DIMS[0],
+	depth=IMAGE_DIMS[2], classes=len(mlb.classes_),
+	finalAct="sigmoid")
+ 
+# initialize the optimizer
+opt = Adam(lr=INIT_LR, decay=INIT_LR / EPOCHS)
+```
+
+`SmallerVGGNet.build`로 `소형VGGNet`모델을 빌드합니다. 이 때 `finalAct="sigmoid"`는 이 모델로 다중 라벨 분류를 수행할 것임을 나타냅니다.
```
Comment (mike2ox) [inline 35_케라스를 이용한 다중 라벨 분류.md]: 문장 앞에 `위 코드에서` 를 추가하면 어떨까요.

File: 35_케라스를 이용한 다중 라벨 분류.md (hunk 5 above)
Comment (mike2ox) [reply 35_케라스를 이용한 다중 라벨 분류.md]: 이 부분은 띄어쓴 곳도 있고 붙여쓴 곳도 있는걸 처음 리뷰한 이후에 확인했습니다. 그대로 띄어 쓰셔도 무방할 것 같습니다.

**Comment by MijeongJeon** [comment]
@fuzzythecat 연준님! 이런 의역 너무 좋습니다. 원문과 비교해봤을때 훨씬 더 자연스럽게 읽혀서 좋네요.
세부 리뷰 곧 작성하겠습니다. 

+ 코드 내부 주석부분 번역 부탁드릴게요!

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 8]
```diff
@@ -0,0 +1,835 @@
...
+# train the network
+print("[INFO] training network...")
+H = model.fit_generator(
+	aug.flow(trainX, trainY, batch_size=BS),
+	validation_data=(testX, testY),
+	steps_per_epoch=len(trainX) // BS,
+	epochs=EPOCHS, verbose=1)
+```
+
+`model.compile`로 모델을 컴파일할 때, **범주형 교차 엔트로피**대신 **이진 교차 엔트로피**를 사용합니다.
```
Comment (MijeongJeon) [inline 35_케라스를 이용한 다중 라벨 분류.md]:
```suggestion
`model.compile`로 모델을 컴파일할 때, **범주형 교차 엔트로피(categorical cross-entropy)**대신 **이진 교차 엔트로피(binary cross-entropy)**를 사용합니다.
```

File: 35_케라스를 이용한 다중 라벨 분류.md [hunk 9]
```diff
@@ -0,0 +1,835 @@
+## 케라스를 이용한 다중 라벨 분류(Multi-label classification with Keras)
+[원문 링크](https://www.pyimagesearch.com/2018/05/07/multi-label-classification-with-keras/)
+> 본 튜토리얼에서는 하나의 CNN을 사용해 옷의 종류와 색을 모두 분류하는 다중 라벨 분석을 다룹니다.
```
Comment (MijeongJeon) [inline 35_케라스를 이용한 다중 라벨 분류.md]:
```suggestion
> 본 튜토리얼에서는 하나의 CNN을 사용해 옷의 종류와 색을 동시에 분류하는 다중 라벨 분석을 다룹니다.
```
